demos/linear_mixed_system/linear_mixed_system.py.rst

```diff
@@ -0,0 +1,328 @@
+
+in which the first line contains integration over fluid domain, second, fluid-structure interface, and third, structure domain. The following notions are used:
+    * :math:`\eta` - free surface deviation
+    * :math:`\phi` - fluid flow potential
+    * :math:`\rho_0` - structure density (in fluid density units)
+    * :math:`\lambda` - first Lame constant (material parameter)
+    * :math:`\mu` - second Lame constant (material parameter)
+    * :math:`{\bf X}` - structure displacement
+    * :math:`{\bf U}` - structure velocity
+    * :math:`e_{ij} = \frac{1}{2} \bigl( \frac{\partial \tilde{X}_j }{ \partial x_i } + \frac{ \partial \tilde{X}_i }{ \partial x_j } \bigr)` - linear strain tensor; :math:`i`, :math:`j` denote vector components
+    * :math:`{\mathrm d} S_f` - integration element over fluid free surface
+    * :math:`{\mathrm d} S_s` - integration element over structure-fluid interface
+    * :math:`{\mathrm d} V_F` - integration element over fluid domain
+    * :math:`{\mathrm d} V_S` - integration element over structure domain
+
+After numerous manipulations (described in detail in :cite:`Salwa:2017`) and evaluation of individual variations, the time-discrete equations, with symplectic Euler scheme, that we would like to implement in Firedrake, are:
+
+.. math::
+
+    \begin{align}
+    & \int v \phi^{n+1} \, {\mathrm d} S_f = \int v (\phi^n - \Delta t \eta^n) \, {\mathrm d} S_f \\
+    & \int \rho_0 {\bf v} \cdot {\bf U}^{n+1} \, {\mathrm d} V_S \underline{ + \int {\bf n} \cdot {\bf v} \, \phi^{n+1} \, {\mathrm d} S_s} = \rho_0 \int {\bf v} \cdot {\bf U}^n \, {\mathrm d} V_S \nonumber\\ 
+    & \hspace{5cm}  - \Delta t \int \left( \lambda \nabla \cdot {\bf v} \nabla \cdot {\bf X}^n + \mu \frac{\partial X^n_j}{\partial x_i}  ( \frac{\partial v_i}{\partial x_j}  + \frac{\partial v_j}{\partial x_i} ) \right) \, {\mathrm d} V_S \underline{ + \int {\bf n} \cdot {\bf v} \, \phi^n \, {\mathrm d} S_s }
+    \\
+    & \int \nabla v \cdot \nabla \phi^{n+1} \, {\mathrm d} V_F \underline{ - \int v {\bf n} \cdot {\bf U}^{n+1} \, {\mathrm d} S_s }= 0 \\ %\hspace{1cm} (+ \text{Dirichlet BC at } \partial \Omega_f)\\
+    & \int v \eta^{n+1} \, {\mathrm d} S_f = \int v \eta^n \, {\mathrm d} S_f + \Delta  t \int \nabla v \cdot \nabla \phi^{n+1} \, {\mathrm d} V_F   \underline{ - \Delta t \int v {\bf n} \cdot {\bf U}^{n+1}\, {\mathrm d} S_s }\\
+    & \int {\bf v} \cdot {\bf X}^{n+1} \, {\mathrm d} V_S = \int {\bf v} \cdot ( {\bf X}^n + \Delta t {\bf U}^{n+1} ) \, {\mathrm d} V_S \, .
+    \end{align}
+
+The underlined terms are the coupling terms. Note that the first equation for :math:`\phi` at the free surface is solved on the free surface only, the last equation for :math:`{\bf X}` in the structure domain, while others in both domains. Moreover, the second and third equations for :math:`\phi` and :math:`{\bf U}` need to be solved simultaneously. Geometry of the system with initial condition is shown below.
+
```

Comment:
"while others in both domains" is ungrammatical. Perhaps "while the others are solved in bot domains".